src/bun.js/api/bun/subprocess.zig

```diff
@@ -2275,17 +2297,36 @@ pub const Subprocess = struct {
                     );
                 }
             },
+            .internal => |data| {
+                IPC.log("Received IPC internal message from child", .{});
+                node_cluster_binding.handleInternalMessagePrimary(this.globalThis, this, data);
+            },
         }
     }
 
     pub fn handleIPCClose(this: *Subprocess) void {
-        this.ipc_data = null;
+        IPClog("Subprocess#handleIPCClose\n", .{});
         this.updateHasPendingActivity();
+        const ok = this.ipc_data != null;
+        if (ok) this.ipc().iimh.deinit();
+        this.ipc_data = null;
+
+        const this_jsvalue = this.this_jsvalue;
+        this_jsvalue.ensureStillAlive();
+        if (this.on_disconnect_callback.trySwap()) |callback| {
+            _ = callback.call(this.globalThis, this_jsvalue, &.{
```

Comment:
Does this need to be wrapped in a event_loop.enter() and event_loop.exit()? What happens if a Promise is awaited inside the disconnect callback?